egs/wenetspeech/ASR/prepare.sh

```diff
@@ -98,47 +99,106 @@ if [ $stage -le 4 ] && [ $stop_stage -ge 4 ]; then
 fi
 
 if [ $stage -le 5 ] && [ $stop_stage -ge 5 ]; then
-  log "Stage 5: Split L subset into ${num_splits} pieces (may take 30 minutes)"
+  log "Stage 5: Split S subset into ${num_splits} pieces"
+  split_dir=data/fbank/S_split_${num_splits}
+  if [ ! -f $split_dir/.split_completed ]; then
+    lhotse split $num_splits ./data/fbank/cuts_S_raw.jsonl.gz $split_dir
+    touch $split_dir/.split_completed
+  fi
+fi
+
+if [ $stage -le 6 ] && [ $stop_stage -ge 6 ]; then
+  log "Stage 6: Split M subset into ${num_splits} piece"
+  split_dir=data/fbank/M_split_${num_splits}
+  if [ ! -f $split_dir/.split_completed ]; then
+    lhotse split $num_splits ./data/fbank/cuts_M_raw.jsonl.gz $split_dir
+    touch $split_dir/.split_completed
+  fi
+fi
+
+if [ $stage -le 7 ] && [ $stop_stage -ge 7 ]; then
+  log "Stage 7: Split L subset into ${num_splits} pieces"
   split_dir=data/fbank/L_split_${num_splits}
   if [ ! -f $split_dir/.split_completed ]; then
     lhotse split $num_splits ./data/fbank/cuts_L_raw.jsonl.gz $split_dir
     touch $split_dir/.split_completed
   fi
 fi
 
-if [ $stage -le 6 ] && [ $stop_stage -ge 6 ]; then
-  log "Stage 6: Compute features for L"
+if [ $stage -le 8 ] && [ $stop_stage -ge 8 ]; then
+  log "Stage 8: Compute features for S"
   python3 ./local/compute_fbank_wenetspeech_splits.py \
+    --training-subset S \
     --num-workers 20 \
     --batch-duration 600 \
     --start 0 \
     --num-splits $num_splits
 fi
 
-if [ $stage -le 7 ] && [ $stop_stage -ge 7 ]; then
-  log "Stage 7: Combine features for L"
-  if [ ! -f data/fbank/cuts_L_50.jsonl.gz ]; then
-    pieces=$(find data/fbank/L_split_50 -name "cuts_L.*.jsonl.gz")
-    lhotse combine $pieces data/fbank/cuts_L_50.jsonl.gz
+if [ $stage -le 9 ] && [ $stop_stage -ge 9 ]; then
+  log "Stage 9: Compute features for M"
+  python3 ./local/compute_fbank_wenetspeech_splits.py \
+    --training-subset M \
+    --num-workers 20 \
+    --batch-duration 600 \
+    --start 0 \
+    --num-splits $num_splits
+fi
+
+if [ $stage -le 10 ] && [ $stop_stage -ge 10 ]; then
+  log "Stage 10: Compute features for L"
+  python3 ./local/compute_fbank_wenetspeech_splits.py \
+    --training-subset L \
+    --num-workers 20 \
+    --batch-duration 600 \
+    --start 0 \
+    --num-splits $num_splits
+fi
+
+if [ $stage -le 11 ] && [ $stop_stage -ge 11 ]; then
+  log "Stage 11: Combine features for S"
+  if [ ! -f data/fbank/cuts_S.jsonl.gz ]; then
+    pieces=$(find data/fbank/S_split_1000 -name "cuts_S.*.jsonl.gz")
+    lhotse combine $pieces data/fbank/cuts_S.jsonl.gz
   fi
 fi
 
-if [ $stage -le 8 ] && [ $stop_stage -ge 8 ]; then
-  log "Stage 8: Compute fbank for musan"
+if [ $stage -le 12 ] && [ $stop_stage -ge 12 ]; then
+  log "Stage 12: Combine features for M"
+  if [ ! -f data/fbank/cuts_M.jsonl.gz ]; then
```

Comment:
You don't need to combine them, I think.
For large datasets, the combining process is slow and the resulting file is large.

You can use
https://github.com/k2-fsa/icefall/blob/a31207f5b3670f5eb0287d24e87bce9e324ed254/egs/librispeech/ASR/pruned_transducer_stateless3/gigaspeech.py#L45-L64
as an example